### ISSUE TITLE
Discover More control is not programmatically a button

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#25586]
## Steps to reproduce
With a VoiceOver / Switch Control / Voice Control running, navigate to the "Discover More" button under the "Thought-provoking stories" and try to activate it while observing the behavior

### Expected behavior
1. VoiceOver should announce it as a button
2. Double-tapping with VoiceOver should activate the control
3. Switch Control should include the button in its auto-focus order (highlight it)
4. Voice Control should be able to call it's on-screen name to activate it

### Actual behavior
1. VoiceOver does not announce any role for this static text 
2. Double-tapping with VoiceOver does not do anything
3. Switch Control does not include the button in its auto-focus order (it does not highlight it)
4. Voice Control does not react on calling its on-screen name

In both cases, users would have to use workarounds like explore-by-touch with VoiceOver and grid with Switch Control or Voice Control

### Device & build information
* Device: iPhone 15 Pro, iOS 18.3.2
* Build version: 9000 (52434)
### Notes
Attachments: 

![Image](https://github.com/user-attachments/assets/2008ca35-a692-40d2-bd7c-fe4c6954907b)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-11732)

**Comment by anna-yeddi** [comment]
![Image](https://github.com/user-attachments/assets/bc088c84-8a57-413b-b7ba-ed053de19bd2) after highlighting the middle card (with the Greanpeace story, in this case) the Switch Control highlights the group, skipping the "Discover More" control

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Verified as fixed on v138(53890), with iPhone 15 (17.6) and iPad Air 11' (18.2). 

“Discover more” is announced as button.
